Country code conversion review

mousebaiker opened the pull request:
Please review the latest commits related to country code conversion.

Don't scroll the conversion map all the way down straight away. Give it some attention as I was generating it with macros and might have screwed up somewhere.

Additionally, all the flags including the ones for provinces should be visible now. The world maps still don't show these provinces as well as there's no aggregation in the list for these fine-grained countries.

__Note:__ I pushed straight to a dev branch, my changes got clustered together with the previous ones and it's a bit hard to review everything right now. Sorry about that. Just cancel the merge request if you think everything is fine. Next time I will make a separate branch for a new feature and make a pull request to the dev branch.

mousebaiker committed:
Refactor for automatic country processing.

This allows for automatic common and unique country extraction, easy country counting
and no manual HTML editing.

UladBohdan committed:
Merge pull request #1 from mousebaiker/master

Refactor for automatic country processing.

mousebaiker committed:
Add full country name to ISO code conversion.

This feature allows to store country data in a human-readable way and then convert it to
easier to use country codes.

Flags work now, expectw for provinces, which need to be handled separatly.

mousebaiker committed:
Display flags for provinces

UladBohdan commented on countrymap.js:
```diff
@@ -0,0 +1,251 @@
+'Guyana': 'GY',
+'Haiti': 'HT',
+'Heard Island and McDonald Islands': 'HM',
+'Holy See (Vatican City State)': 'VA',
```
I was unsure about the spellings of some countries and if google charts can recognize them - such as `Holy See (Vatican City State)`, `Côte d\'Ivoire`, `Åland Islands` etc., but I did verify they work just fine.
Good job!

mousebaiker replied:
Thank you for review and actual verification. I did not check if they worked myself; just believed that Wikipedia listing will have them named in a proper manner and decided that they countries are not the most popular and can be checked when somebody actually visits them.

UladBohdan commented:
Not merging this.
Will be merged once the feature is complete.

mousebaiker committed:
Aggregate provinces of a single country to a single table row.

As some countries are visualized on a region level, we would like these
regions to be grouped together in a single entry.

Additionally, this patch takes care of visual discrepancy between these
fine-grained region codes and simple country name. As a side effect,
names in data.js can be specified by either full name of by 2 letter country code.

UladBohdan committed:
Merge pull request #4 from UladBohdan/aggregation

Aggregate provinces of a single country to a single table row.
